README.md

```diff
@@ -114,26 +114,64 @@ The cache contains Linux x64 binaries of all packages that are used during a def
 - *to use a certain commit for any of the inputs*: use flag `--override-input`, e.g. 
   ```sh
   nix develop --override-input all-cabal-hashes "github:commercialhaskell/all-cabal-hashes/f4b3c68d6b5b128503bc1139cfc66e0537bccedd"
-  ```
-- if you plan to upstream your modifications to `ghc.nix`, don't forget to run the formatter (`nix fmt`)
+  ``` 
+  this is not yet support in `flake-compat` mode, you will have to manually set the version in the `flake.nix` by appending 
+  `/your-commit-hash` to the input you want to change, then running `nix flake lock --update-input input-you-want-to-update`. 
+  Of course you can also just manually pass your own `nixpkgs` version to the `shell.nix`, this will override the one 
+  provided by the flake.
+- if you plan to upstream your modifications to `ghc.nix`, don't forget to run the formatter using `nix fmt`
 
 ## Flake support
 
-`ghc.nix` now also has basic flake support, `nixpkgs` and `nixpkgs-unstable` are pinned in the flake inputs
-and the cabal hashes are pinned in the toplevel flake (not via the flake inputs because that is currently not 
-possible in an idiomatic way) 
-
-To format all nix code in this repo, run `nix fmt`, to enter a development shell, run `nix develop`.
-
-## direnv
-
-With nix-direnv support, it is possible to make [`direnv`](https://github.com/direnv/direnv/) load `ghc.nix`
+`ghc.nix` now also has basic flake support, `nixpkgs`, `nixpkgs-unstable` and the `cabal-hashes` are pinned in the flake inputs.
+
+- To change the settings of the `devShell` to your liking, just adjust the `userSettings` attribute-set in the top-level flake.
+- To format all nix code in this repo, run `nix fmt`, to enter a development shell, run `nix develop`.
+
+## Legacy nix-commands support
```

Comment:
I think, an example of a local/custom `shell.nix` file would be helpful

Reply:
I think I do not understand what you mean here, sorry.

Reply:
Okay, I moved into my attic and found this (Who said that sloppiness wouldn't pay off one day :wink: ):

```nix
import ./ghc.nix/default.nix {
#  bootghc = "ghc865";
#  withIde = true;
#  version = "8.8";
  withHadrianDeps = true;
#  cores = 8;
#  withDocs = false;
}
```

This is the content of a very old `shell.nix`. Would this work with your current changes? (Roughly, this `shell.nix` is pretty old, so some details might have changed.)

If yes, IMHO it could be useful to paste the contents of such a file into `README.md`.

Reply:
Yeah, this is of course totally possible, except you'd call the shell.nix in our case. Passing an attrset to the shell.nix directly corresponds to passing attributes with --arg

Reply:
fixed.